docker-app/qfieldcloud/core/signals.py

```diff
@@ -1,8 +1,45 @@
 from axes.signals import user_locked_out
 from django.dispatch import receiver
 from qfieldcloud.core.exceptions import TooManyLoginAttemptsError
+from qfieldcloud.core.models import OrganizationMember, User, Project
+from django.db.models.signals import pre_delete
 
 
 @receiver(user_locked_out)
 def raise_permission_denied(*args, **kwargs):
     raise TooManyLoginAttemptsError()
+
+
+@receiver(pre_delete, sender=User)
+def update_project_owner_on_member_delete(sender, instance, **kwargs):
+    """
+    Update project ownership when an organization member is deleted.
+    """
+
+    fallback_user = None
+
+    projects_owned_by_user = Project.objects.filter(owner=instance)
+
+    if projects_owned_by_user.exists():
+        fallback_user = projects_owned_by_user.first().owner
+
+    if not fallback_user:
+        try:
+            org_membership = OrganizationMember.objects.get(member=instance)
+            fallback_user = org_membership.organization.organization_owner
+
+        except OrganizationMember.DoesNotExist:
+            raise ValueError(
+                f"No OrganizationMember found for user: {instance.username}"
+            )
+
+    if fallback_user:
+        OrganizationMember.objects.filter(created_by=instance).update(
+            created_by=fallback_user
+        )
+        OrganizationMember.objects.filter(updated_by=instance).update(
+            updated_by=fallback_user
+        )
+
+    else:
+        print("No fallback user found. Skipping updates.")
```

Comment:
I am not sure what is it about project ownership, so the docs and function name are saying "Update project ownership"? You don't update the owner, you update the created/update by *with* the owner.

Also you are updating all the organization members with the same `fallback_user`, even if the `fallback_user` has nothing to do with the particular project.

"o1" is owned by "u2"
"o2" is owned by "u3"
"u1" is member of "o1"
"u1" is member of "o2"
"o1" is owner of "p1"
"o2" is owner of "p2"
"u1" added member in both "o1" and "o2".

"u1" is deleted

Then "u2" or "u3" will be used for all `OrgranizationMembers` that were added by "u1" in both "o1" and "o2", even though "u2" is not member of "o2" and "u3" is not member of "o1", because of `projects_owned_by_user.first().owner`, or using the `first()`  to be more precise.


Note there is also a print, please use `logging` module whenever needed.

Reply:
Btw, having a second look on this, I see two more problems:

1)
```
    projects_owned_by_user = Project.objects.filter(owner=instance)

    if projects_owned_by_user.exists():
        fallback_user = projects_owned_by_user.first().owner
```
If you filter by owner that equals `instance` and then you get the first result owner, then `fallback_user == instance` no?

2) 
```
        try:
            org_membership = OrganizationMember.objects.get(member=instance)
            fallback_user = org_membership.organization.organization_owner

        except OrganizationMember.DoesNotExist:
            raise ValueError(
                f"No OrganizationMember found for user: {instance.username}"
            )
```

What if you have more than one membership? The `get(member=instance)` will fail with `[OrganizationMember.MultipleObjectsReturned](https://docs.djangoproject.com/en/5.1/ref/models/class/#django.db.models.Model.MultipleObjectsReturned) and will raise a `ValueError`.

Reply:
As we said, we do not implement signal for now